### ISSUE TITLE
Clean up the Reporting Tools

### ISSUE THREAD

**Issue by john-science** [terrapower/armi#793]
## Original Problem Statement

At some point, @youngmit and @chris10mckenz built some new reporting tools. Awesome! However, the project was left unfinished. We we have the original/old reporting tools:

* `armi/bookkeeping/report/data.py`
* `armi/bookkeeping/report/html.py`
* `armi/bookkeeping/report/reportingUtils.py`
* `armi/bookkeeping/report/reportInterface.py`

But we also have the "new" reporting tools:

* `armi/bookkeeping/report/newReports.py`
* `armi/bookkeeping/report/newReportUtils.py`
* `armi/bookkeeping/report/report.js`
* `armi/bookkeeping/report/style.css`

Obviously, leaving the word "new" on those files indefinitely seems silly. This is a confusing state of affairs for our user base. We should combine these tools into one set, and simplify our API.

## Progress Update

The layout of the reporting tools is now much simpler:

* `armi/bookkeeping/report/data.py`
* `armi/bookkeeping/report/reportingUtils.py`
* `armi/bookkeeping/report/reportInterface.py`
* `armi/utils/reportPlotting.py`

But there are still some common concerns I hear about ARMI's reporting tools:

1. Some of the log info or plots are not used, and never have been.
2. Some plots are either broken, or too fragile.
3. Some tools are hyper-specific to only work with Fast or Hex-based, pin-type reactors.
    * Separate these out and make them easily identified.
4. ARMI should just provide TOOLS for reporting things, in a `ReportInterface`. But leave all the detailed choices of what reactor-specific reports and plots should be created to downstream subclasses of the `ReportInterface`.

**Comment by ntouran** [comment]
Ah, _reporting_. Yeah we should definitely merge and clean as appropriate. As we clean this up, it might make sense to have armi define some data structures that plugins can pass around with intent to report, and leave users the option to render reports with whatever reporting technology they want, given a bunch of packaged report data structures.

**Comment by john-science** [comment]
As a first step to the "merge and clean the reporting tools", I have moved all the "newReport" tools into the "reporting" package. 

https://github.com/terrapower/armi/pull/867

Honestly, I don't know why the new stuff wasn't there to begin with.

**Comment by john-science** [comment]
I also note that `armi/utils/reportPlotting.py` imports extensively from the "old" armi reports tools.

I was also able to find downstream projects using:

* `from armi.bookkeeping import report`
  * `report.setData()`
  * ` report.CORE_RESOURCES`
  * `report.ECONOMICS`
  * `report.COST_ASSUMPTIONS`
  * `report.CORE_AVG_ENRICH`
  * `report.COST_ASSUMPTIONS`
  * `report.STRUCT_RESOURCES`
  * `report.CORE_RESOURCES`
  * `report.FUEL_CYC_COST`
* `from armi.utils import reportPlotting`
  * `reportPlotting.createPlotMetaData`
  * `reportPlotting.plotAxialProfile`
* `from armi.bookkeeping.report import reportingUtils`
  * `reportingUtils.getInterfaceStackSummary()`

So, actually, outside of ARMI, I think the direction import usage of our reporting tools are pretty minor.  Mostly, we just have to support he above few things.  And most of the work will be reorganzing our internal usage.

**Comment by john-science** [comment]
Also, happily, I couldn't find anyone outside of ARMI using the "newReport" tools. So, we can change that dubious name to something else with ease.

**Comment by john-science** [comment]
Okay, this situation has gone on too long.

I can find no one anywhere using "newReports.py" or "newReportUtils.py", so the easiest solution is clearly to delete them. They will remain in Git history if someone wants them.  And since they are undocumented, I don't really know what they do or why they exist. Shame.

**Comment by albeanth** [comment]
dialogue from @john-science to the reporting utils files 😄 

![grim-reaper-bad-news](https://github.com/user-attachments/assets/a040d8b9-3309-44bb-81eb-3de9be322e0e)

**Comment by keckler** [comment]
I would like to see this implemented, if that counts for anything.

**Comment by john-science** [comment]
> I would like to see this implemented, if that counts for anything.

Well, SOME project out there is telling me to code freeze.  #squint

**Comment by albeanth** [comment]
You should(?) be clear to remove these, I think! We can talk more offline though, if you'd like.

**Comment by jakehader** [comment]
@john-science / @keckler - Since reporting is so specific to a model or what a user wants from some analysis, would it make sense to plugin-ify this work so it's not lost to time in git commits?

**Comment by albeanth** [comment]
@john-science some of our downstream users have told us that what gets spit out here
https://github.com/terrapower/armi/blob/f86ef61964cab1be8064468627906d85519788eb/armi/bookkeeping/report/reportingUtils.py#L743-L773

is actually flat out incorrect. So I think it's prudent that at least the old reporting tools get removed.

**Comment by john-science** [comment]
Some progress made on this recently; https://github.com/terrapower/armi/pull/2098

**Comment by john-science** [comment]
## I have put out a call for information:

1. Are there plots or logged reports in ARMI that are broken and can be removed?
2. Are there plots or logged reports in ARMI that should be moved to to specific TerraPower projects that they are designed for?

## Responses

### @drewj-tp said: 

I very rarely need / use the generated plots.

* assembly plots can get wild if you have more than 5 assembly types
  * I have directories with 130 generated assembly plots
  * @keckler says:
    *  "I think the assemblies plots are very useful, but hyper-specific and less-than-useful in multiple ways"
    * TLDR; maybe move them out of ARMI.
* if the axial meshing is too complicated, the plots become unreadable because of all the y-axis labels
  * this is pushing the bounds of usability
* the BOL assembly mass summary table loses all sense of usefulness once you hit some threshold of assemblies
* delete the code timing plot (just the plot)
  * @keckler says "I agree that the code timing plot is useless."
* the comprehensive reporting print out should either be axed or moved internal. I think this is something where every org would have their own Opinions 

### @mgjarrett said:

I will say I have never seen or heard of anybody using the "maxPD" or the "moves":
 
https://github.com/terrapower/armi/blob/81427e487126371cfc44c09371ff59e5f8411aa3/armi/utils/reportPlotting.py#L92-L101
https://github.com/terrapower/armi/blob/81427e487126371cfc44c09371ff59e5f8411aa3/armi/utils/reportPlotting.py#L110

@keckler says "I have used the "moves" plot." "Never heard of the maxPD plot"

### @bsculac said:

* does anyone use the spent fuel pool report?
  * doesn't look like we populate it anyway

**Comment by john-science** [comment]
So far, excluding a redesign of the assembly plots, there is some low-hanging fruit:
 
* Axe the [SFP report](https://github.com/terrapower/armi/blob/9a8a49c6e9d589b0a57e921ecd57a57e1a722a8a/armi/bookkeeping/report/reportInterface.py#L145-L167) (for situations where people aren't modeling the SFP)
* Axe the [Code Timing plot](https://github.com/terrapower/armi/blob/81427e487126371cfc44c09371ff59e5f8411aa3/armi/utils/codeTiming.py#L234-L315)
* Axe the [maxPD plot](https://github.com/terrapower/armi/blob/81427e487126371cfc44c09371ff59e5f8411aa3/armi/utils/reportPlotting.py#L92-L101)
* Move the the comprehensive reporting print out of ARMI, and leave a simpler version in ARMI.
* For [this plot]( https://github.com/terrapower/armi/blob/9a8a49c6e9d589b0a57e921ecd57a57e1a722a8a/armi/utils/plotting.py#L685), print the y-axis at fixed increments
